### PR TITLE
Hide IPP references in Transaction settings when Cash on Delivery is disabled

### LINKED ISSUE #6972: Unclear description when enabling Auth & Capture
### Describe the bug
In the Payments -> settings we have "Issue an authorization on checkout, and capture later" checkbox and it has description 
> Charge must be captured on the order details screen within 7 days of authorization, otherwise the authorization and order will be canceled. The setting is not applied to In-Person Payments (please note that In-Person Payments should be captured within 2 days of authorization).

Given that my account doesn't have in-Person Payments enabled it seems confusing that it's mentioned. If I'm not aware of what in-Person payments are, I might be tempted to bail out of Auth & Capture.

### Screenshots

image

### PR DESCRIPTION
**Pull request by mgascam**
Fixes #6972 

#### Changes proposed in this Pull Request
* ~Expose `available_payment_gateways` in the response of `settings` endpoint~.  EDIT: Changed the approach, now check if Cash on Display is available when calculating `is_card_present_eligible` in the settings.
* Migrate `ManualCaptureControl` component to TSX. (note: I attempted migration of other components used by this one but I'll defer this migration to a dedicated issue since migrating the data layer qualifies for a dedicated issue).

**Before**

Screenshot 2023-11-06 at 16 30 18

**After**

Screenshot 2023-11-06 at 16 31 08

#### Testing instructions

* You'll need a merchat account based on a IPP enabled country: US, CA, GB.
* As a merchant, ensure that `Cash on Delivery` is enabled in `Payments > Settings > All Payment Methods`.
* Navigate to Payments > Settings > WooPayments.
* Scroll down to the Transactions section and check that the description below `Issue an authorization on checkout, and capture later` is: 
```
Charge must be captured on the order details screen within 7 days of authorization, otherwise the authorization and order will be canceled. The setting is not applied to In-Person Payments (please note that In-Person Payments should be captured within 2 days of authorization).
``` 
* Disable `Cash on Delivery`.
* Navigate to Payments > Settings > WooPayments.  Scroll down to the Transactions section and check that the description below `Issue an authorization on checkout, and capture later` is: 
```
Charge must be captured on the order details screen within 7 days of authorization, otherwise the authorization and order will be canceled.
``` 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7638` or branch name `update/6972-unclear-description-when-enabling-auth-capture` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/6972-unclear-description-when-enabling-auth-capture) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 3c6a6c0cc6f9f9c76642505d6491de01406ecc0b
- Build time: 2023-11-09 15:18:33 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by KarlisJ — APPROVED** [review]
Code looks good, tests well.

File: includes/admin/class-wc-rest-payments-settings-controller.php [hunk 1]
```diff
@@ -467,6 +467,7 @@ public function get_settings(): WP_REST_Response {
 		return new WP_REST_Response(
 			[
 				'enabled_payment_method_ids'          => $enabled_payment_methods,
+				'available_payment_gateways'          => WC()->payment_gateways->get_available_payment_gateways(),
```
Comment (kalessil) [inline includes/admin/class-wc-rest-payments-settings-controller.php]: Not sure if this is going to work, but can we consider something like `'is_card_present_eligible'            => $this->wcpay_gateway->is_card_present_eligible() && isset( WC()->payment_gateways()->payment_gateways()['cod'] ),`?

**Review by kalessil — CHANGES_REQUESTED** [review]
There is a way to simplify the fix (I didn't dive into the flag usage, though). 
TS migration and new test cases (if it makes sense with the suggested change) is a welcome change, though.

**Comment by kalessil** [comment]
@aheckler @mgascam would it make sense to link 'In-person Payments' wording in the text we are touching to https://woo.com/in-person-payments/?

File: includes/admin/class-wc-rest-payments-settings-controller.php (hunk 1 above)
Comment (mgascam) [reply includes/admin/class-wc-rest-payments-settings-controller.php]: Good point @kalessil . I checked the usages of the `is_card_present_eligible` flag. Apart from `ManualCaptureControl` component, it is used to conditionally show the `Card Readers` entry in the Payments menu: https://github.com/Automattic/woocommerce-payments/blob/92525c2a637bf592ec412bb0a979ab91862575d1/includes/admin/class-wc-payments-admin.php#L399

So I removed the 'available_payment_gateways' in favor of your suggestion in this commit: https://github.com/Automattic/woocommerce-payments/pull/7638/commits/e30e49732d875f7a461400fc09abf7f3e0283401. For more context, I was wrongly assuming that the `is_card_present_eligible` was used as a feature flag, but now I see it is not the case. Thanks!

**Comment by mgascam** [comment]
> @aheckler @mgascam would it make sense to link 'In-person Payments' wording in the text we are touching to https://woo.com/in-person-payments/?

I believe it makes sense. I'm adding it in this commit https://github.com/Automattic/woocommerce-payments/pull/7638/commits/d7f145d9f38dd212ba64d319994022269dd41883, it can be easily reverted if we decide otherwise.

Screenshot 2023-11-08 at 14 09 30

**Review by kalessil — APPROVED** [review]
LGTM (code review only)!

File: includes/admin/class-wc-rest-payments-settings-controller.php (hunk 1 above)
Comment (kalessil) [reply includes/admin/class-wc-rest-payments-settings-controller.php]: Thanks! I'm pre-approving, but there are some failing tests, and the changelog seems missing.

**Review by kalessil — APPROVED** [review]
Did a testing round, works like  acharm!

@mgascam, can you please add a space before `The setting is not applied to [In-Person Payments(opens in a new tab)](https://woo.com/in-person-payments/) (please note that In-Person Payments should be captured within 2 days of authorization).`?

We are a good to merge the PR with the requested change - I marking it as ready to merge.

**Pull request merged**